src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -585,6 +585,12 @@ private bool canTailCall(CORINFO_METHOD_STRUCT_* callerHnd, CORINFO_METHOD_STRUC
                 throw new RequiresRuntimeJitException(nameof(fIsTailPrefix));
             }
 
+            MethodDesc exactCallee = HandleToObject(exactCalleeHnd);
+            if (exactCallee != null && exactCallee.RequireSecObject)
+            {
+                return false;
+            }
+
             return false;
```

Comment:
Isn't these 2 returns redundant?

Reply:
Yes, they are and I have intentionally put it there. I've mentioned it in the PR description.

Reply:
I see, my bad.